pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -290,35 +208,44 @@ func libImageName(registry string, lang language, tag string) string {
 	return fmt.Sprintf(imageFormat, registry, lang, tag)
 }
 
+func (w *Webhook) isPodEligible(pod *corev1.Pod) bool {
+	return ShouldInject(pod)
+}
+
+// isEnabledInNamespace indicates if Single Step Instrumentation is enabled for
+// the namespace in the cluster
+func (w *Webhook) isEnabledInNamespace(namespace string) bool {
+	return IsEnabledInNamespace(namespace)
+}
+
 func (w *Webhook) inject(pod *corev1.Pod, _ string, _ dynamic.Interface) (bool, error) {
 	if pod == nil {
 		return false, errors.New(metrics.InvalidInput)
 	}
+
 	injectApmTelemetryConfig(pod)
 
-	if w.isEnabledInNamespace(pod.Namespace) {
```

Comment:
This was originally lifted into `isPodEligible`, tested, and then refactored to call `ShouldInject` which we were calling later in the call stack in `extractLibInfo`. (prme-1)